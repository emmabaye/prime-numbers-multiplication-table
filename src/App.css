.app {
  text-align: center;
  margin-left: 50px;
  margin-right: 50px;
  margin-top: 100px;
}

@media (max-width: 665px) {
  .app {
    margin-left: 0px;
    margin-right: 0px;
  }
}
 
