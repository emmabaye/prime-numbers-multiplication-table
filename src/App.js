import React from 'react';
import './App.css';
import { getMultiplicationTableArray, getPrimeNumbers } from './utils/primeNumberUtils';

function App() {
  const tableArray = getMultiplicationTableArray(getPrimeNumbers(10));

  return (
    <div className="app">
      <h3>Multiplication Table of Prime Numbers</h3>
      <table data-testid="multiplication-table" className="table table-striped table-bordered">
        <thead>
          <tr>
        {tableArray[0].map((cell, cellIndex) => {
          return <th scope="col" key={cellIndex}>{cell}</th>;
        })}
         </tr>
        </thead>
        <tbody>
        {tableArray.map((row, rowIndex) => {
          return(rowIndex === 0) ? null : (
            <tr key={rowIndex}>
              {
                row.map((cell, cellIndex) => {
                  return (cellIndex === 0 ) ? <th scope="row"  key={cellIndex}>{cell}</th> : <td  key={cellIndex}>{cell}</td>
                })
              }
            </tr>
            );
        })}
        </tbody>
      </table>
    </div>
  );
}

export default App;
