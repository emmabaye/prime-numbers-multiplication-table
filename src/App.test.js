import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('should take a snapshot of table', () => {
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});

test('should render table', () => {
  const { getByTestId } = render(<App />);

  expect(getByTestId('multiplication-table')).toBeTruthy();
  expect(getByTestId('multiplication-table').className).toBe("table table-striped table-bordered");
});

