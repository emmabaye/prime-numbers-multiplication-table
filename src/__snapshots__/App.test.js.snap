// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of table 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <h3>
      Multiplication Table of Prime Numbers
    </h3>
    <table
      class="table table-striped table-bordered"
      data-testid="multiplication-table"
    >
      <thead>
        <tr>
          <th
            scope="col"
          />
          <th
            scope="col"
          >
            2
          </th>
          <th
            scope="col"
          >
            3
          </th>
          <th
            scope="col"
          >
            5
          </th>
          <th
            scope="col"
          >
            7
          </th>
          <th
            scope="col"
          >
            9
          </th>
          <th
            scope="col"
          >
            11
          </th>
          <th
            scope="col"
          >
            13
          </th>
          <th
            scope="col"
          >
            17
          </th>
          <th
            scope="col"
          >
            19
          </th>
          <th
            scope="col"
          >
            23
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            scope="row"
          >
            2
          </th>
          <td>
            4
          </td>
          <td>
            6
          </td>
          <td>
            10
          </td>
          <td>
            14
          </td>
          <td>
            18
          </td>
          <td>
            22
          </td>
          <td>
            26
          </td>
          <td>
            34
          </td>
          <td>
            38
          </td>
          <td>
            46
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            3
          </th>
          <td>
            6
          </td>
          <td>
            9
          </td>
          <td>
            15
          </td>
          <td>
            21
          </td>
          <td>
            27
          </td>
          <td>
            33
          </td>
          <td>
            39
          </td>
          <td>
            51
          </td>
          <td>
            57
          </td>
          <td>
            69
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            5
          </th>
          <td>
            10
          </td>
          <td>
            15
          </td>
          <td>
            25
          </td>
          <td>
            35
          </td>
          <td>
            45
          </td>
          <td>
            55
          </td>
          <td>
            65
          </td>
          <td>
            85
          </td>
          <td>
            95
          </td>
          <td>
            115
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            7
          </th>
          <td>
            14
          </td>
          <td>
            21
          </td>
          <td>
            35
          </td>
          <td>
            49
          </td>
          <td>
            63
          </td>
          <td>
            77
          </td>
          <td>
            91
          </td>
          <td>
            119
          </td>
          <td>
            133
          </td>
          <td>
            161
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            9
          </th>
          <td>
            18
          </td>
          <td>
            27
          </td>
          <td>
            45
          </td>
          <td>
            63
          </td>
          <td>
            81
          </td>
          <td>
            99
          </td>
          <td>
            117
          </td>
          <td>
            153
          </td>
          <td>
            171
          </td>
          <td>
            207
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            11
          </th>
          <td>
            22
          </td>
          <td>
            33
          </td>
          <td>
            55
          </td>
          <td>
            77
          </td>
          <td>
            99
          </td>
          <td>
            121
          </td>
          <td>
            143
          </td>
          <td>
            187
          </td>
          <td>
            209
          </td>
          <td>
            253
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            13
          </th>
          <td>
            26
          </td>
          <td>
            39
          </td>
          <td>
            65
          </td>
          <td>
            91
          </td>
          <td>
            117
          </td>
          <td>
            143
          </td>
          <td>
            169
          </td>
          <td>
            221
          </td>
          <td>
            247
          </td>
          <td>
            299
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            17
          </th>
          <td>
            34
          </td>
          <td>
            51
          </td>
          <td>
            85
          </td>
          <td>
            119
          </td>
          <td>
            153
          </td>
          <td>
            187
          </td>
          <td>
            221
          </td>
          <td>
            289
          </td>
          <td>
            323
          </td>
          <td>
            391
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            19
          </th>
          <td>
            38
          </td>
          <td>
            57
          </td>
          <td>
            95
          </td>
          <td>
            133
          </td>
          <td>
            171
          </td>
          <td>
            209
          </td>
          <td>
            247
          </td>
          <td>
            323
          </td>
          <td>
            361
          </td>
          <td>
            437
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            23
          </th>
          <td>
            46
          </td>
          <td>
            69
          </td>
          <td>
            115
          </td>
          <td>
            161
          </td>
          <td>
            207
          </td>
          <td>
            253
          </td>
          <td>
            299
          </td>
          <td>
            391
          </td>
          <td>
            437
          </td>
          <td>
            529
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
